packages/jest-snapshot/src/mock_serializer.js

```diff
@@ -17,12 +17,19 @@ export const serialize = (
   refs: Refs,
   printer: Printer,
 ): string => {
-  const mockObject = {
-    calls: val.mock.calls,
-    name: val.getMockName(),
-  };
+  const indentationNext = indentation + config.indent;
 
-  return printer(mockObject, config, indentation, depth, refs);
+  return (
```

Comment:
It feels really convoluted to do this. @pedrottimark is it necesarry to just change the header of the object?

Reply:
FYI, a solution for a **built-in** plugin is to call `printObjectProperties` for example:

https://github.com/facebook/jest/blob/master/packages/pretty-format/src/plugins/asymmetric_matcher.js#L40-L58

**But** `AsymmetricMatcher` will soon change **not** to do that for the reason that I missed when I first read https://github.com/facebook/jest/issues/4835#issue-271013580

> which will make impossible to accidentally match with an object.

Our currently proposed serialization matches instance constructed from ordinary class:

```js
class MockFunction {
  constructor(calls = [], name = 'jest.fn()') {
    this.calls = calls;
    this.name = name;
  }
}
```

When I read **Mock Functions** page to borrow an example for alternative, a **question** hit me:

Should the serialization also include `instances` property?

http://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockinstances

Reply:
I originally had `instances` as well, but it didn't really provide anything of value.

https://github.com/facebook/jest/commit/3514516a29300afdebcd0a834a3239d2cd675423